ipatests/test_integration/test_upgrade.py

```diff
@@ -5,13 +5,96 @@
 """
 Module provides tests to verify that the upgrade script works.
 """
-
 import base64
+import configparser
+import os
+import io
+
 from cryptography.hazmat.primitives import serialization
+import pytest
+
+from ipaplatform.paths import paths
 from ipapython.dn import DN
+from ipapython.ipautil import template_str
+from ipaserver.install import bindinstance
+from ipaserver.install.sysupgrade import STATEFILE_FILE
 from ipatests.test_integration.base import IntegrationTest
 from ipatests.pytest_ipa.integration import tasks
 
+# old template without comments for testing
+# and "dnssec-validation no"
+OLD_NAMED_TEMPLATE = """
+options {
+        listen-on-v6 {any;};
+        directory "$NAMED_VAR_DIR"; // the default
+        dump-file               "${NAMED_DATA_DIR}cache_dump.db";
+        statistics-file         "${NAMED_DATA_DIR}named_stats.txt";
+        memstatistics-file      "${NAMED_DATA_DIR}named_mem_stats.txt";
+        tkey-gssapi-keytab "$NAMED_KEYTAB";
+        pid-file "$NAMED_PID";
+        dnssec-enable yes;
+        dnssec-validation no;
+        bindkeys-file "$BINDKEYS_FILE";
+        managed-keys-directory "$MANAGED_KEYS_DIR";
+        $INCLUDE_CRYPTO_POLICY
+};
+
+logging {
+        channel default_debug {
+                file "${NAMED_DATA_DIR}named.run";
+                severity dynamic;
+                print-time yes;
+        };
+};
+
+include "$RFC1912_ZONES";
+include "$ROOT_KEY";
+
+/* WARNING: This part of the config file is IPA-managed.
+ * Modifications may break IPA setup or upgrades.
+ */
+dyndb "ipa" "$BIND_LDAP_SO" {
+        uri "ldapi://%2fvar%2frun%2fslapd-$SERVER_ID.socket";
+        base "cn=dns, $SUFFIX";
```

Comment:
You removed space in the 'base' in the template. Do you want to remove it here as well?